packages/shared/src/hdf5-utils.ts

```diff
@@ -117,6 +117,7 @@ export function compoundOrCplxType<T extends DType>(
   fields: Record<string, T>,
 ): CompoundType<T> | ComplexType {
   const { r, i } = fields;
+  // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
   if (r && isNumericType(r) && i && isNumericType(i)) {
```

Comment:
This is the false positive I was talking about due to `noUncheckedIndexedAccess` not being enabled.

`fields.<whatever>` always gives a `string` instead of `string | undefined`, which is why the `r &&` and `i &&` checks are supposedly unnecessary.

I will keep in mind that we should really strive to enable `noUncheckedIndexedAccess` eventually.